### PR TITLE
Hayden branch

### PR DESCRIPTION
**Pull request by haydenreich**
- Directory structure changes
 - Made hatServer into a package
 - Refactored algorithm to pull data from Mongo database rather than the parsed .csv file
 - Added utility to seed the database from the .csv files

### CODE REVIEW

File: hatServer/controllers.py [hunk 1]
```diff
@@ -1,14 +1,21 @@
...
-from mongoengine import *
+from flask import Flask, request, Response
+from flask import render_template, url_for, redirect, send_from_directory
+from flask import send_file, make_response, abort
+
+from hatServer.forms import LoginForm
+
 import os
 
+<<<<<<< HEAD:hatServer/controllers.py
```
Comment (jepetty) [inline hatServer/controllers.py]: Oh, we've got an unresolved merge conflict!

File: hatServer/config.py [hunk 2]
```diff
@@ -0,0 +1,4 @@
+WTF_CSRF_ENABLED = True
```
Comment (jepetty) [inline hatServer/config.py]: I don't know if these two things are still necessary now that we're not using WTForms... Did you need to add them to change the configuration of hatServer?

File: hatServer/controllers.py [hunk 3]
```diff
@@ -1,14 +1,21 @@
-from flask import Flask, render_template, request
-from flask_mongoengine import MongoEngine
-from mongoengine import *
+from flask import Flask, request, Response
+from flask import render_template, url_for, redirect, send_from_directory
+from flask import send_file, make_response, abort
+
+from hatServer.forms import LoginForm
```
Comment (jepetty) [inline hatServer/controllers.py]: We'll need to remove this line when we remove the WTForms stuff

File: hatServer/controllers.py [hunk 4]
```diff
@@ -21,6 +28,21 @@ class Groups(Document):
     group_name = StringField(required=True)
     members = ListField(ReferenceField(Students))
 """
+
+@app.route('/')
+@app.route('/about')
+def basic_pages(**kwargs):
+    return make_response(open('hatServer/templates/index.html').read())
+
+@app.route('/login', methods=['GET', 'POST'])
```
Comment (jepetty) [inline hatServer/controllers.py]: I do believe this method is also outdated without the forms stuff.

File: hatServer/forms.py [hunk 5]
```diff
@@ -0,0 +1,8 @@
+from flask_wtf import Form
```
Comment (jepetty) [inline hatServer/forms.py]: This file can also be removed.

File: hatServer/sortingHat/sortingHat.py [hunk 6]
```diff
@@ -0,0 +1,238 @@
...
+            swapThemBitches(group, True, groups)
+            if len(group.members) < MIN_SIZE:
+                swapThemBitches(group, False, groups)
+
+    for group in matched:
+        matched[group] = group.members
+
+    return matched
+
+def dumbledore():
```
Comment (jepetty) [inline hatServer/sortingHat/sortingHat.py]: Can we have a comment describing what this method does?

**Review by jepetty — CHANGES_REQUESTED** [review]
Some of my comments were about full removal of the WTForms stuff. I don't know if it's already in master, but I'm happy to open a pull request myself tackling that if you want to keep the focus on configuration of the hatServer.

**Comment by jepetty** [comment]
Also, did you mean to remove all of the angular routing and controller stuff? I thought we were going to stick with the angular routing and not do WTForms?

**Comment by haydenreich** [comment]
I accidentally wiped the static directory when I moved it under hatServer. So now routing should work again. Oops

**Review by jepetty — APPROVED** [review]

**Pull request merged**